### PR TITLE
do Part5

### PR DESCRIPTION
**Pull request by KottV**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -407,9 +429,22 @@ bool SpaceShip::dock()
...
 {
-    ++planetNum;
+    int loop = 0;
+    while (loop <= loopCount)
+    {
+        ++loop;
+        if ( loop == planetNum )
+        {
+            std::cout << "Planet "<<planetNum<< " Say bye!" << std::endl;
+            this->totalLoops += loop;
```
Comment (matkatmusic) [inline main.cpp]: add spaces around your `<<` so your code is more readable.

`this` hasn't been introduced yet.   Don't do more work than you are asked to do.
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -526,6 +561,45 @@ void Body::alarmOverDust(float time)
...
+
+    float step = 1.0f / ( static_cast<float>(cval) + 1.0f );
+    
+    if ( pval < cval )
+    {
+        led.Brightness = 0;
+
+        for ( led.Num = 0; led.Num <= cval; ++led.Num )
+        {
+            led.Brightness+=step;
```
Comment (matkatmusic) [inline main.cpp]: add space around the `+=` to make your code more readable
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -526,6 +561,45 @@ void Body::alarmOverDust(float time)
...
+    else
+    {
+        while ( led.Num > cval )
+        {
+            led.Brightness = 0;
+            led.set();
+            --led.Num;
+        }
+
+        ( cval !=  0  ? led.Brightness = 1 : 0 );
```
Comment (matkatmusic) [inline main.cpp]: When you write ternary expressions, you can make them more readable by writing them like this:
```
led.brightness = (expr ? true statement : false statement);
```
as in:
```
led.brightness = (cval != 0 ? 1 : 0 );
```
- [ ] revise
- [ ] `Brightness` is a member variable, and member variable names need to start with a `lowerCaseLetter`

File: main.cpp [hunk 4]
```diff
@@ -526,6 +561,45 @@ void Body::alarmOverDust(float time)
...
+        for ( led.Num = cval; led.Num >= 0; --led.Num )
+        {
+            led.set();
+            led.Brightness-=step;
```
Comment (matkatmusic) [inline main.cpp]: add space around your `-=`
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -526,6 +561,45 @@ void Body::alarmOverDust(float time)
...
+    {
+        while ( led.Num > cval )
+        {
+            led.Brightness = 0;
+            led.set();
+            --led.Num;
+        }
+
+        ( cval !=  0  ? led.Brightness = 1 : 0 );
+        for ( led.Num = cval; led.Num >= 0; --led.Num )
```
Comment (matkatmusic) [inline main.cpp]: member variable names need to begin with a lowerCase letter.
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -276,6 +279,25 @@ struct Body
...
 
+struct Knob
+{
+    int pvalue{0};
+    int cvalue{0};
+
+    struct Led
+    {
+        int Num = 0;
+        float Brightness = 0;
```
Comment (matkatmusic) [inline main.cpp]: member variable names need to start with a lower case letter.
- [ ] revise

File: main.cpp [hunk 7]
```diff
@@ -276,6 +279,25 @@ struct Body
...
+            std::cout << Num << " " << Brightness << std::endl;
+        }
+    };
+    
+    int setValue(int, int);
+};
+
 struct MonitorController
 {
     DAC DACLeft;
```
Comment (matkatmusic) [inline main.cpp]: member variable names need to start with a lowercase letter
- [ ] revise

File: main.cpp [hunk 8]
```diff
@@ -407,9 +429,22 @@ bool SpaceShip::dock()
...
+        ++loop;
+        if ( loop == planetNum )
+        {
+            std::cout << "Planet "<<planetNum<< " Say bye!" << std::endl;
+            this->totalLoops += loop;
+            return loop;
+        }
+            
+    }
+    this->totalLoops += loop;
```
Comment (matkatmusic) [inline main.cpp]: `this` hasn't been introduced yet.
- [ ] revise

File: main.cpp [hunk 9]
```diff
@@ -576,40 +650,22 @@ bool MonitorController::toggleCrossfeed(bool status)
 int main()
 {
     Example::main();
-    Cat Pusya;
-    SipProvider sipnet;
     SpaceShip Proton;
```
Comment (matkatmusic) [inline main.cpp]: variable names need to start with a lower case letter
- [ ] revise

File: main.cpp [hunk 10]
```diff
@@ -576,40 +650,22 @@ bool MonitorController::toggleCrossfeed(bool status)
...
-    DAC dacLeft;
-    
-    std::cout << "\nSIP trunk status:" << std::endl;
-    sipnet.printStatus ();
-    
-    std::cout << "\n";
-    powerunit.printStatus();
     
-    std::cout << "\nDAC status: " << dacLeft.checkError(440) << std::endl;
+    for ( int i = 2; i < 5; ++i ) Proton.makeLoop(i, 6);
```
Comment (matkatmusic) [inline main.cpp]: Even if your `if/else/for/while/etc` expressions have only one statement, you should still put it on its own line.
```
for( int i = 2; i < 5; ++i )
    proton.makeLoop(i, 6);
```
- [ ] revise

File: main.cpp [hunk 11]
```diff
@@ -576,40 +650,22 @@ bool MonitorController::toggleCrossfeed(bool status)
...
-    << "age: " << Pusya.age << "\n"
-    << "gender: "<< ( Pusya.gender == 0 ? "Girl" : "Boy" ) << std::endl;
-    
-    Pusya.mew (3);
-    
-    // fill balance
-    sipnet.balance = 100;
-    std::cout << "\nSIP trunk status:" << std::endl;
-    sipnet.printStatus ();
+    for ( int i=0;i < 3; ++i )
```
Comment (matkatmusic) [inline main.cpp]: add space around your `=` and `;` to make your code more readable
- [ ] revise

File: main.cpp [hunk 12]
```diff
@@ -576,40 +650,22 @@ bool MonitorController::toggleCrossfeed(bool status)
...
+    {
+        std::cout << "Set volume: ";
+        std::cin >> volume.cvalue;
```
Comment (matkatmusic) [inline main.cpp]: Don't use `cin` in these projects
- [ ] revise

File: main.cpp [hunk 13]
```diff
@@ -576,40 +650,22 @@ bool MonitorController::toggleCrossfeed(bool status)
...
+        if ( volume.cvalue == volume.pvalue ) continue;
```
Comment (matkatmusic) [inline main.cpp]: Even if your `if/else/for/while/etc` expressions have only one statement, you should still put it on its own line.
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Please take a look at the course coding style guide found in the Readme.MD

File: main.cpp (hunk 3 above)
Comment (KottV) [reply main.cpp:591]: does it work with "int"?

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**